src/sentry/api/endpoints/organization_region.py

```diff
@@ -0,0 +1,73 @@
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint, control_silo_endpoint
+from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.api.permissions import SentryPermission
+from sentry.models.organizationmapping import OrganizationMapping
+from sentry.models.organizationmembermapping import OrganizationMemberMapping
+from sentry.types.region import get_region_by_name
+from sentry.utils.types import Any
+
+
+class OrganizationRegionEndpointPermissions(SentryPermission):
+    # Although this permission set is a bit weird, we need to have
+    # project:read for integration auth tokens, org:ci for org auth tokens
```

Comment:
It isn't that strange for a token with access to a project to be able to read the region that the containing organization is located in.

Reply:
Discussed this during stand-up, but this ended up being the minimal set of requirements for our custom integrations tokens to be able to use this endpoint as intended.